packages/x-tree-view/src/internals/plugins/useTreeViewExpansion/useTreeViewExpansion.test.tsx

```diff
@@ -0,0 +1,244 @@
+import { expect } from 'chai';
+import { spy } from 'sinon';
+import { describeTreeView } from 'test/utils/tree-view/describeTreeView';
+import { UseTreeViewExpansionSignature } from '@mui/x-tree-view/internals';
+import { act, fireEvent } from '@mui-internal/test-utils';
+
+/**
+ * All tests related to keyboard navigation (e.g.: expanding using "Enter" and "ArrowRight")
+ * are located in the `useTreeViewKeyboardNavigation.test.tsx` file.
```

Comment:
This file does not exist yet of course, but you get the idea :laughing: 
The reasoning is that when you expand with the keyboard, there are a lot of focus / expansion / selection considerations to take into account, and have all of those handled by the keyboard navigation plugin and tests is simpler IHMO.